staging/src/k8s.io/client-go/transport/transport_test.go

```diff
@@ -168,37 +173,114 @@ func TestNew(t *testing.T) {
+					},
+					GetKey: func() ([]byte, error) {
+						return []byte(keyData), nil
+					},
+				},
+			},
+		},
+		"cert callback error": {
+			TLS:     true,
+			TLSCert: true,
+			TLSErr:  true,
+			Config: &Config{
+				TLS: TLSConfig{
+					CAData: []byte(rootCACert),
+					GetCert: func() ([]byte, error) {
+						return nil, errors.New("GetCert failure")
+					},
+					GetKey: func() ([]byte, error) {
+						return []byte(keyData), nil
+					},
+				},
+			},
+		},
+		"cert data overrides empty callback result": {
+			TLS:     true,
+			TLSCert: true,
+			Config: &Config{
+				TLS: TLSConfig{
+					CAData: []byte(rootCACert),
+					GetCert: func() ([]byte, error) {
+						return nil, nil
+					},
+					CertData: []byte(certData),
+					GetKey: func() ([]byte, error) {
+						return []byte(keyData), nil
+					},
+				},
+			},
+		},
+		"callbacks return nothing": {
+			TLS:     true,
+			TLSCert: true,
+			Config: &Config{
+				TLS: TLSConfig{
+					CAData: []byte(rootCACert),
+					GetCert: func() ([]byte, error) {
+						return nil, nil
+					},
+					GetKey: func() ([]byte, error) {
+						return nil, nil
+					},
+				},
+			},
+		},
 	}
 	for k, testCase := range testCases {
-		transport, err := New(testCase.Config)
-		switch {
-		case testCase.Err && err == nil:
-			t.Errorf("%s: unexpected non-error", k)
-			continue
-		case !testCase.Err && err != nil:
-			t.Errorf("%s: unexpected error: %v", k, err)
-			continue
-		}
+		t.Run(k, func(t *testing.T) {
+			rt, err := New(testCase.Config)
+			switch {
+			case testCase.Err && err == nil:
+				t.Fatal("unexpected non-error")
+			case !testCase.Err && err != nil:
+				t.Fatalf("unexpected error: %v", err)
+			}
+			if testCase.Err {
+				return
+			}
 
-		switch {
-		case testCase.Default && transport != http.DefaultTransport:
-			t.Errorf("%s: expected the default transport, got %#v", k, transport)
-			continue
-		case !testCase.Default && transport == http.DefaultTransport:
-			t.Errorf("%s: expected non-default transport, got %#v", k, transport)
-			continue
-		}
+			switch {
+			case testCase.Default && rt != http.DefaultTransport:
+				t.Fatalf("got %#v, expected the default transport", rt)
+			case !testCase.Default && rt == http.DefaultTransport:
+				t.Fatalf("got %#v, expected non-default transport", rt)
+			}
 
-		// We only know how to check TLSConfig on http.Transports
-		if transport, ok := transport.(*http.Transport); ok {
+			// We only know how to check TLSConfig on http.Transports
+			transport := rt.(*http.Transport)
 			switch {
 			case testCase.TLS && transport.TLSClientConfig == nil:
-				t.Errorf("%s: expected TLSClientConfig, got %#v", k, transport)
-				continue
+				t.Fatalf("got %#v, expected TLSClientConfig", transport)
 			case !testCase.TLS && transport.TLSClientConfig != nil:
-				t.Errorf("%s: expected no TLSClientConfig, got %#v", k, transport)
-				continue
+				t.Fatalf("got %#v, expected no TLSClientConfig", transport)
+			}
+			if testCase.TLS {
```

Comment:
Here and for TLSCert, do like you do above:
```golang
if !testCase.TLS{
  return
}
```

Reply:
Done